apps/server/src/modules/common-cartridge/common-cartridge-client/card-client/mapper/card-response.mapper.spec.ts

```diff
@@ -0,0 +1,216 @@
+								text: 'Rich text content',
+								inputFormat: 'Markdown',
+							}) as RichTextElementResponse,
+							createMockElement(
+								'element-unknown',
+								'UNKNOWN_TYPE' as ContentElementType,
+								{}
+							) as CardResponseElementsInner,
+						],
+						visibilitySettings: { publishedAt: '2024-10-01T12:00:00Z' },
+						timestamps: {
+							lastUpdatedAt: '2024-10-01T11:00:00Z',
+							createdAt: '2024-10-01T10:00:00Z',
+							deletedAt: faker.date.recent().toString(),
+						},
+					},
+				],
+			};
+
+			const result = CardResponseMapper.mapToCardListResponseDto(mockCardListResponse);
+			expect(result).toBeDefined();
+			expect(result.data).toHaveLength(1);
+
+			const cardResponseDto = result.data[0];
+			expect(cardResponseDto.id).toBe('card-1');
+			expect(cardResponseDto.title).toBe('Card 1');
+			expect(cardResponseDto.height).toBe(200);
+			expect(cardResponseDto.visibilitySettings.publishedAt).toBe('2024-10-01T12:00:00Z');
+			expect(cardResponseDto.timeStamps.lastUpdatedAt).toBe('2024-10-01T11:00:00Z');
+
+			expect(cardResponseDto.elements).toHaveLength(8);
+			expect(cardResponseDto.elements[0].type).toBe(ContentElementType.COLLABORATIVE_TEXT_EDITOR);
+			expect(cardResponseDto.elements[1].type).toBe(ContentElementType.DELETED);
+			expect(cardResponseDto.elements[2].type).toBe(ContentElementType.SUBMISSION_CONTAINER);
+			expect(cardResponseDto.elements[3].type).toBe(ContentElementType.DRAWING);
+			expect(cardResponseDto.elements[4].type).toBe(ContentElementType.EXTERNAL_TOOL);
+			expect(cardResponseDto.elements[5].type).toBe(ContentElementType.FILE);
+			expect(cardResponseDto.elements[6].type).toBe(ContentElementType.LINK);
+			expect(cardResponseDto.elements[7].type).toBe(ContentElementType.RICH_TEXT);
+		});
+
+		it('should handle unknown element types without breaking', () => {
+			const mockCardListResponse: CardListResponse = {
+				data: [
+					{
+						id: 'card-2',
+						title: 'Card 2',
+						height: 150,
+						elements: [
+							createMockElement(
+								'element-unknown',
+								'UNKNOWN_TYPE' as ContentElementType,
+								{}
+							) as CardResponseElementsInner,
+						],
+						visibilitySettings: { publishedAt: faker.date.past().toISOString() },
+						timestamps: {
+							lastUpdatedAt: faker.date.past().toISOString(),
+							createdAt: faker.date.past().toISOString(),
+							deletedAt: faker.date.recent().toString(),
+						},
+					},
+				],
+			};
+
+			const result = CardResponseMapper.mapToCardListResponseDto(mockCardListResponse);
+			expect(result).toBeDefined();
+			expect(result.data).toHaveLength(1);
+
+			const cardResponseDto = result.data[0];
+			expect(cardResponseDto.id).toBe('card-2');
+			expect(cardResponseDto.title).toBe('Card 2');
+			expect(cardResponseDto.height).toBe(150);
+			expect(cardResponseDto.elements).toHaveLength(0);
+		});
+		it('should return an empty list of elements when CardResponse has no elements', () => {
```

Comment:
check testing doc. When statements should be added to a describe block to provide more descritptions of the test case
```suggestion
    describe('when CardResponse has no elements')
		it('should return an empty list of elements', () => {
```

Reply:
Done